From 7ebd50337cd930af9632c089d5e045641a7ed474 Mon Sep 17 00:00:00 2001
From: Allen Xu <b45815@freescale.com>
Date: Wed, 24 Sep 2014 14:57:27 -0500
Subject: [PATCH 0728/5242] ENGR00333047 ARM: dts: add new dts for ecspi on
 imx6dl sabreauto

commit  d3bbb7fe8eeedba0448b9f41ca6b7eebdce4cbc4 from
https://source.codeaurora.org/external/imx/linux-imx.git

new dts file to support ecspi on imx6dl sabreauto board.

Signed-off-by: Allen Xu <b45815@freescale.com>
(cherry picked and merge from commit 243e501dbec2b62148c6a470226abced76438eac)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                   |    1 +
 arch/arm/boot/dts/imx6dl-sabreauto-ecspi.dts |   42 ++++++++++++++++++++++++++
 2 files changed, 43 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6dl-sabreauto-ecspi.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index c4f5f79..bddbe15 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -409,6 +409,7 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6dl-rex-basic.dtb \
 	imx6dl-riotboard.dtb \
 	imx6dl-sabreauto.dtb \
+	imx6dl-sabreauto-ecspi.dtb \
 	imx6dl-sabreauto-gpmi-weim.dtb \
 	imx6dl-sabrelite.dtb \
 	imx6dl-sabresd.dtb \
diff --git a/arch/arm/boot/dts/imx6dl-sabreauto-ecspi.dts b/arch/arm/boot/dts/imx6dl-sabreauto-ecspi.dts
new file mode 100644
index 0000000..23d362c
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dl-sabreauto-ecspi.dts
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6dl-sabreauto.dts"
+
+&ecspi1 {
+	pinctrl-assert-gpios = <&gpio5 4 GPIO_ACTIVE_LOW>;
+	status = "okay";
+};
+
+/* remove the #if 0 after flexcan is ready */
+#if 0
+&flexcan2 {
+	/* max7310_c on i2c3 is gone */
+	status = "disabled";
+};
+#endif
+
+&i2c3 {
+	/* pin conflict with ecspi1 */
+	status = "disabled";
+};
+
+&uart3 {
+	/* the uart3 depends on the i2c3, so disable it too. */
+	status = "disabled";
+};
+
+&usbh1 {
+	/* max7310_b on i2c3 is gone */
+	status = "disabled";
+};
+
+&usbotg {
+	dr_mode = "peripheral";
+	status = "okay";
+};
-- 
1.7.9.5

